controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -588,6 +586,24 @@ func (r *ReconcileHyperConverged) updateHyperConverged(request *common.HcoReques
 		return false, err
 	}
 
+	if request.Dirty {
```

Comment:
now the fakeClient we are using for unit tests finally behaves like the real APIServer
so we can finally revert to the designed implementation.
See: https://github.com/kubernetes-sigs/controller-runtime/pull/2259